llama_stack/apis/datasets/datasets.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from typing import Any, Dict, List, Optional, Protocol
+
+from llama_models.llama3.api.datatypes import URL
+
+from llama_models.schema_utils import json_schema_type, webmethod
+
+from pydantic import BaseModel, Field
+
+from llama_stack.apis.common.type_system import ParamType
+
+
+@json_schema_type
+class DatasetSchema(BaseModel):
+    columns: Dict[str, ParamType]
+
+
+@json_schema_type
+class DatasetDef(BaseModel):
+    identifier: str = Field(
+        description="A unique name for the dataset",
+    )
+    dataset_schema: DatasetSchema = Field(
```

Comment:
now wondering if we should just inline the `columns` property of schema into the `Def` itself?